tests/neo.UnitTests/SmartContract/Native/UT_StdLib.cs

```diff
@@ -0,0 +1,161 @@
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Neo.SmartContract;
+using Neo.SmartContract.Native;
+using Neo.VM;
+using Neo.VM.Types;
+using System.Collections.Generic;
+using System.Numerics;
+using System.Threading;
+
+namespace Neo.UnitTests.SmartContract.Native
+{
+    [TestClass]
+    public class UT_StdLib
+    {
+        [TestInitialize]
+        public void Init()
+        {
+            TestBlockchain.InitializeMockNeoSystem();
+            Thread.Sleep(1000); // TODO: We should not wait
+        }
+
+        [TestMethod]
+        public void TestBinary()
+        {
+            var data = System.Array.Empty<byte>();
+
+            CollectionAssert.AreEqual(data, StdLib.Base64Decode(StdLib.Base64Encode(data)));
+            CollectionAssert.AreEqual(data, StdLib.Base58Decode(StdLib.Base58Encode(data)));
+
+            data = new byte[] { 1, 2, 3 };
+
+            CollectionAssert.AreEqual(data, StdLib.Base64Decode(StdLib.Base64Encode(data)));
+            CollectionAssert.AreEqual(data, StdLib.Base58Decode(StdLib.Base58Encode(data)));
+            Assert.AreEqual("AQIDBA==", StdLib.Base64Encode(new byte[] { 1, 2, 3, 4 }));
+            Assert.AreEqual("2VfUX", StdLib.Base58Encode(new byte[] { 1, 2, 3, 4 }));
+        }
+
+        [TestMethod]
+        public void TestItoaAtoi()
+        {
+            Assert.AreEqual("1", StdLib.Itoa(BigInteger.One, 10));
+            Assert.AreEqual("1", StdLib.Itoa(BigInteger.One, 16));
+            Assert.AreEqual("-1", StdLib.Itoa(BigInteger.MinusOne, 10));
+            Assert.AreEqual("f", StdLib.Itoa(BigInteger.MinusOne, 16));
+            Assert.AreEqual(-1, StdLib.Atoi("-1", 10));
+            Assert.AreEqual(1, StdLib.Atoi("+1", 10));
+            Assert.AreEqual(-1, StdLib.Atoi("ff", 16));
+            Assert.ThrowsException<System.FormatException>(() => StdLib.Atoi("a", 10));
+            Assert.ThrowsException<System.ArgumentOutOfRangeException>(() => StdLib.Atoi("a", 11));
```

Comment:
can we add these cases as well? StdLib.Atoi("g", 16)  StdLib.Atoi("FF", 16)    StdLib.Itoa(1_000_000_000, 16)

Reply:
done